chapters/01-preface.Rmd

```diff
@@ -0,0 +1,334 @@
+# Preface {#preface}
+
+Computational skills are essential in the modern research environment, which
+increasingly involves large datasets and high performance computing,
+along with increasing demand that scientific studies be reproducible and
+open. With these modern demands, researchers across nearly all disciplines often
+need to create, use, and share software. Most introductions to programming or
+software engineering focus on developing commercial applications, however,
+researchers needs are largely related to exploring problems and answering
+questions. This book covers the fundamentals of creating software as part of
+your research cycle whether you work alone or as part of a team. We wrote this
+book with three audiences in mind:
+
+1. **Learners in a class setting** who would be taking this course in a more
+formal class or workshop setting that is lead by an instructor or instructors.
+Much of the book is written with this reader in mind.
+2. **Independent learners** who would be going through the material and learning
+on their own. An independent learner can for the most part follow the book as if
+they were a learner in a class setting, with some modifications that we
+highlight and describe where necessary.
+3. **Instructors** who would use this book as the basis for a course or workshop
+of their own. Throughout the book there are sections dedicated to this reader
+that give information or instructions on how to use or modify the material for
+the instructor's specific context.
+
+## Introduction for learners
+
+This primary aim of this book is to teach you the tools, skills, and knowledge
+necessary to making research software that is open, reproducible, and
+sustainable. For researchers and data scientists who can write functions to
+create programs that are hundreds of lines long and who want to be more
+productive and have more confidence in their results, this book provides a
+pragmatic, tools-based introduction to research software engineering. 
+For researchers whose tasks include building software packages, this book will
+help train you to be a [research software engineer][rse]. Unlike courses aimed
+at computer scientists and professional software developers, this book uses data
+analysis as the motivating example and assumes that the learner's ultimate goal
+is to answer questions rather than ship products.
```

Comment:
I'm wondering whether we're hitting appropriate topics here, given the intended audience (as well as discouraging the unintended audience): 
- "ship products" is a niche bit of jargon for software developers: is this something researchers/undergrads would understand?
- I think we're still missing the link between why writing packages is meaningful in the context of RSE (again, this is from the perspective of an undergrad)